app/src/main/java/org/mozilla/focus/ext/Context.kt

```diff
@@ -0,0 +1,17 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.ext
+
+import android.content.Context
+
+// Extension functions for the Context class
+
+val Context.appVersionName: String?
+    get() {
+        val packageInfo = packageManager.getPackageInfo(packageName, 0)
+        return packageInfo.versionName
+    }
+
+fun Context.dp(px: Int): Int = (px * resources.displayMetrics.density).toInt()
```

Comment:
While brief, I don't think this method should be encapsulated on Context, e.g. you can convert px to dp with just a DisplayMetrics instance. I wouldn't be super opposed if you were to have an ext fn on Resources which calls the DisplayMetrics ext fn, especially because I don't think I ever remember what DisplayMetrics is.

There's also the greater issue that Context as way too much responsibility as it is. :d

Reply:
At first I wanted to add this to Int. I saw an example where they added this as an extension property of int and using it looked like this:
> 8.dp

However this used the system's display metrics and this didn't work well. So I needed a Context to get the actual display metrics. That's how it ended up here. Inspired by Anko:
https://github.com/Kotlin/anko/blob/d5a526512b48c5cd2e3b8f6ff14b153c2337aa22/anko/library/static/commons/src/Dimensions.kt#L38

Adding it to DisplayMetrics might be the correct way. But If I have to call context.resources.displayMetrics.dp(..) in the code everywhere then I can almost do the conversion myself without a helper. :)

As mentioned above I can move the padding into the layout and can remove the helper method for now. This gives us more time to think about this. :)